src/msg/async/AsyncConnection.cc

```diff
@@ -744,6 +749,7 @@ void AsyncConnection::cleanup() {
     delete delay_state;
     delay_state = NULL;
   }
+  protocol.reset();
```

Comment:
```
2020-02-13T12:03:54.142 INFO:tasks.ceph.mon.a.smithi102.stderr:*** Caught signal (Segmentation fault) **
2020-02-13T12:03:54.142 INFO:tasks.ceph.mon.a.smithi102.stderr: in thread 7fa128cdf700 thread_name:fn_monstore
2020-02-13T12:03:54.143 INFO:tasks.ceph.mgr.x.smithi107.stderr:2020-02-13T12:03:54.141+0000 7faa7be28040 -1 WARNING: all dangerous and experimental features are enabled.
2020-02-13T12:03:54.143 INFO:tasks.ceph.mgr.y.smithi102.stderr:*** Caught signal (Segmentation fault) **
2020-02-13T12:03:54.143 INFO:tasks.ceph.mgr.y.smithi102.stderr: in thread 7faae27ef040 thread_name:ceph-mgr
2020-02-13T12:03:54.145 INFO:tasks.ceph.mgr.x.smithi107.stderr:2020-02-13T12:03:54.142+0000 7faa7be28040 -1 WARNING: all dangerous and experimental features are enabled.
2020-02-13T12:03:54.145 INFO:tasks.ceph.mon.a.smithi102.stderr: ceph version 15.1.0-596-g3be473c (3be473c7d578040af701705444ac5bec35ae4f4a) octopus (rc)
2020-02-13T12:03:54.145 INFO:tasks.ceph.mon.a.smithi102.stderr: 1: (()+0x12dc0) [0x7fa1358b7dc0]
2020-02-13T12:03:54.145 INFO:tasks.ceph.mon.a.smithi102.stderr: 2: (AsyncConnection::is_connected()+0xb) [0x7fa137e11f2b]
2020-02-13T12:03:54.145 INFO:tasks.ceph.mon.a.smithi102.stderr: 3: (PaxosService::dispatch(boost::intrusive_ptr<MonOpRequest>)+0xda) [0x5597fedad61a]
2020-02-13T12:03:54.145 INFO:tasks.ceph.mon.a.smithi102.stderr: 4: (PaxosService::C_RetryMessage::_finish(int)+0x64) [0x5597fed17d94]
2020-02-13T12:03:54.145 INFO:tasks.ceph.mon.a.smithi102.stderr: 5: (C_MonOp::finish(int)+0x49) [0x5597fecbbf79]
2020-02-13T12:03:54.146 INFO:tasks.ceph.mon.a.smithi102.stderr: 6: (Context::complete(int)+0xd) [0x5597fecb933d]
2020-02-13T12:03:54.146 INFO:tasks.ceph.mon.a.smithi102.stderr: 7: (void finish_contexts<std::__cxx11::list<Context*, std::allocator<Context*> > >(CephContext*, std::__cxx11::list<Context*, std::allocator<Context*> >&, int)+0xa5) [0x5597fece15d5]
2020-02-13T12:03:54.146 INFO:tasks.ceph.mon.a.smithi102.stderr: 8: (PaxosService::_active()+0x1cb) [0x5597fedaeebb]
2020-02-13T12:03:54.146 INFO:tasks.ceph.mon.a.smithi102.stderr: 9: (Context::complete(int)+0xd) [0x5597fecb933d]
2020-02-13T12:03:54.146 INFO:tasks.ceph.mon.a.smithi102.stderr: 10: (void finish_contexts<std::__cxx11::list<Context*, std::allocator<Context*> > >(CephContext*, std::__cxx11::list<Context*, std::allocator<Context*> >&, int)+0xa5) [0x5597fece15d5]
2020-02-13T12:03:54.146 INFO:tasks.ceph.mon.a.smithi102.stderr: 11: (Paxos::finish_round()+0x169) [0x5597feda49d9]
2020-02-13T12:03:54.146 INFO:tasks.ceph.mon.a.smithi102.stderr: 12: (Paxos::commit_finish()+0x91c) [0x5597feda755c]
2020-02-13T12:03:54.147 INFO:tasks.ceph.mon.a.smithi102.stderr: 13: (C_Committed::finish(int)+0x45) [0x5597fedab8f5]
2020-02-13T12:03:54.147 INFO:tasks.ceph.mon.a.smithi102.stderr: 14: (Context::complete(int)+0xd) [0x5597fecb933d]
2020-02-13T12:03:54.147 INFO:tasks.ceph.mon.a.smithi102.stderr: 15: (MonitorDBStore::C_DoTransaction::finish(int)+0x98) [0x5597fedab618]
2020-02-13T12:03:54.147 INFO:tasks.ceph.mon.a.smithi102.stderr: 16: (Context::complete(int)+0xd) [0x5597fecb933d]
2020-02-13T12:03:54.147 INFO:tasks.ceph.mon.a.smithi102.stderr: 17: (Finisher::finisher_thread_entry()+0x1a5) [0x7fa137bfb5c5]
2020-02-13T12:03:54.147 INFO:tasks.ceph.mon.a.smithi102.stderr: 18: (()+0x82de) [0x7fa1358ad2de]
2020-02-13T12:03:54.147 INFO:tasks.ceph.mon.a.smithi102.stderr: 19: (clone()+0x43) [0x7fa134657133]
2020-02-13T12:03:54.147 INFO:tasks.ceph.mon.a.smithi102.stderr:2020-02-13T12:03:54.142+0000 7fa128cdf700 -1 *** Caught signal (Segmentation fault) **
2020-02-13T12:03:54.148 INFO:tasks.ceph.mon.a.smithi102.stderr: in thread 7fa128cdf700 thread_name:fn_monstore
```

see http://pulpito.ceph.com/kchai-2020-02-13_11:45:07-rados-wip-kefu-testing-2020-02-13-0025-distro-basic-smithi/

Reply:
Oops, yet another race. Apparently somebody calls this after `cleanup()`.

```cpp
bool AsyncConnection::is_connected() {
  return protocol->is_connected();
}
```